app/assets/javascripts/admin/addon/services/site-setting-change-tracker.js

```diff
@@ -59,40 +63,140 @@ export default class SiteSettingChangeTracker extends Service {
       });
 
+      this.#stopSaving();
+      this.dirtySiteSettings.clear();
+
       if (reload) {
         reload();
       }
     } catch (error) {
       popupAjaxError(error);
-    } finally {
-      this.dirtySiteSettings.forEach((setting) =>
-        setting.set("isSaving", false)
-      );
     }
   }
 
   discard() {
-    this.dirtySiteSettings.forEach((siteSetting) => siteSetting.cancel());
+    this.dirtySiteSettings.forEach((setting) =>
+      setting.buffered.discardChanges()
+    );
+    this.dirtySiteSettings.clear();
+  }
+
+  async confirmChanges(setting) {
+    const settingKey = setting.buffered.get("setting");
+
+    return new Promise((resolve) => {
+      // Fallback is needed in case the setting does not have a custom confirmation
+      // prompt/confirm defined.
+      this.dialog.alert({
+        message: i18n(
+          `admin.site_settings.requires_confirmation_messages.${settingKey}.prompt`,
+          {
+            translatedFallback: i18n(
+              "admin.site_settings.requires_confirmation_messages.default.prompt"
+            ),
+          }
+        ),
+        buttons: [
+          {
+            label: i18n(
+              `admin.site_settings.requires_confirmation_messages.${settingKey}.confirm`,
+              {
+                translatedFallback: i18n(
+                  "admin.site_settings.requires_confirmation_messages.default.confirm"
+                ),
+              }
+            ),
+            class: "btn-primary",
+            action: () => resolve(true),
+          },
+          {
+            label: i18n("no_value"),
+            class: "btn-default",
+            action: () => resolve(false),
+          },
+        ],
+      });
+    });
+  }
+
+  async confirmTransition() {
+    await new Promise((resolve) => {
+      this.dialog.confirm({
+        message: i18n("admin.site_settings.dirty_banner", {
+          count: this.count,
+        }),
+        confirmButtonLabel: "admin.site_settings.save",
+        cancelButtonLabel: "admin.site_settings.discard",
+        didConfirm: async () => {
+          await this.save();
+          resolve(true);
+        },
+        didCancel: () => {
+          this.discard();
+          resolve(false);
+        },
+      });
+    });
+  }
+
+  async configureBackfill(setting) {
+    const key = setting.buffered.get("setting");
+
+    const data = {
+      [key]: setting.buffered.get("value"),
+    };
+
+    const result = await ajax(`/admin/site_settings/${key}/user_count.json`, {
+      type: "PUT",
+      data,
+    });
+
+    const count = result.user_count;
+
+    if (count > 0) {
+      await this.modal.show(SiteSettingDefaultCategoriesModal, {
+        model: {
+          siteSetting: { count, key: key.replaceAll("_", " ") },
+          setUpdateExistingUsers: setting.setUpdateExistingUsers,
+        },
+      });
+    }
+  }
+
+  #startSaving() {
+    this.dirtySiteSettings.forEach((setting) => {
+      setting.set("isSaving", true);
```

Comment:
Since isSaving has been turned into a tracked property, you can directly assign it values and it will auto-update dependent stuff (even computed properties).

```suggestion
      setting.isSaving = true;
```